processing/src/main/java/io/druid/query/topn/types/DoubleTopNColumnSelectorStrategy.java

```diff
@@ -0,0 +1,118 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.topn.types;
+
+import com.google.common.base.Function;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.query.topn.BaseTopNAlgorithm;
+import io.druid.query.topn.TopNParams;
+import io.druid.query.topn.TopNQuery;
+import io.druid.query.topn.TopNResultBuilder;
+import io.druid.segment.Capabilities;
+import io.druid.segment.Cursor;
+import io.druid.segment.DoubleColumnSelector;
+import io.druid.segment.column.ValueType;
+import it.unimi.dsi.fastutil.longs.Long2ObjectMap;
+import it.unimi.dsi.fastutil.longs.Long2ObjectOpenHashMap;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+
+public class DoubleTopNColumnSelectorStrategy implements TopNColumnSelectorStrategy <DoubleColumnSelector, Long2ObjectMap<Aggregator[]>>
+{
+
+  @Override
+  public int getCardinality(DoubleColumnSelector selector)
+  {
+    return TopNColumnSelectorStrategy.CARDINALITY_UNKNOWN;
+  }
+
+  @Override
+  public ValueType getValueType()
+  {
+    return ValueType.DOUBLE;
+  }
+
+  @Override
+  public Aggregator[][] getDimExtractionRowSelector(
+      TopNQuery query, TopNParams params, Capabilities capabilities
+  )
+  {
+    return new Aggregator[0][];
+  }
+
+  @Override
+  public Long2ObjectMap<Aggregator[]> makeDimExtractionAggregateStore()
+  {
+    return new Long2ObjectOpenHashMap<>();
+  }
+
+  @Override
+  public long dimExtractionScanAndAggregate(
```

Comment:
I think it could be a single class, implementing three different interfaces via overloading